File: sample/src/main/java/fr/o80/app/ABTestingApplication.java
package fr.o80.app;

import android.app.Application;

import fr.o80.lib.abtesting.ABPercentConfig;
import fr.o80.lib.abtesting.ABTesting;

/**
 * @author Olivier Perez
 */
public class ABTestingApplication extends Application {

    private ABTesting abTesting;

    @Override
    public void onCreate() {
        super.onCreate();

        // Configure ABTesting
        abTesting = new ABTesting(this);

        ABPercentConfig percent = abTesting.percent(ABTestingConst.ABTESTING_EXAMPLE, ABTestingConst.ABTESTING_EXAMPLE_DEFAULT);
        percent.addCase(0, 33, "A");
        percent.addCase(34, 66, "B");

        // Init values
        abTesting.init();
    }

    public ABTesting getABTesting() {
        return abTesting;
    }
}

File: sample/src/main/java/fr/o80/app/ABTestingApplication.kt
package fr.o80.app

import android.app.Application
import android.util.Log
import fr.o80.lib.abtesting.ABPercentConfig.DefaultCase
import fr.o80.lib.abtesting.ABTesting
import fr.o80.lib.abtesting.abConfiguration

/**
 * @author Olivier Perez
 */
class ABTestingApplication : Application() {

    lateinit var abTesting: ABTesting
        private set

    override fun onCreate() {
        super.onCreate()

        // Configure ABTesting
        abTesting = abConfiguration(this) {
            percentage(ABTestingConst.ABTESTING_EXAMPLE) {
                "A" isIn 1..33
                "B" isIn 34..66
                "default" isFor DefaultCase
            }
        }

        Log.d("", "")
    }
}
